docarray/array/mixins/evaluation.py

```diff
@@ -1,59 +1,147 @@
 import warnings
-from typing import Optional, Union, TYPE_CHECKING, Callable
+from typing import Optional, Union, TYPE_CHECKING, Callable, List, Dict
 
 import numpy as np
+from collections import defaultdict
 
 from docarray.score import NamedScore
 
 if TYPE_CHECKING:
     from docarray import Document, DocumentArray
 
 
+def _evaluate_deprecation(f):
+    """Raises a deprecation warning if the user executes the evaluate function with
+    the old interface and adjust the input to fit the new interface."""
+
+    def func(*args, **kwargs):
+        if len(args) > 1:
+            if not (
```

Comment:
why not applying `positive logic`?

Reply:
I can not check for instanceof(DocumentArray) because it results in a circular dependency.